src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -429,16 +425,16 @@ void ObjectSynchronizer::handle_sync_on_value_based_class(Handle obj, Thread* cu
 // of this algorithm. Make sure to update that code if the following function is
 // changed. The implementation is extremely sensitive to race condition. Be careful.
 
-void ObjectSynchronizer::enter(Handle obj, BasicLock* lock, TRAPS) {
+void ObjectSynchronizer::enter(Handle obj, BasicLock* lock, JavaThread* current) {
   if (obj->klass()->is_value_based()) {
-    handle_sync_on_value_based_class(obj, THREAD);
+    handle_sync_on_value_based_class(obj, current);
   }
 
   if (UseBiasedLocking) {
     if (!SafepointSynchronize::is_at_safepoint()) {
-      BiasedLocking::revoke(obj, THREAD);
+      BiasedLocking::revoke(obj, current);
     } else {
-      BiasedLocking::revoke_at_safepoint(obj);
+      guarantee(false, "VMThread should not be involved with ObjectMonitor");
```

Comment:
Interesting change. Seems out of context for this change.
Since you have "guarantee(false," you can use "fatal(" instead...